workflow/controller/operator.go

```diff
@@ -1338,6 +1338,16 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 		if err != nil {
 			return woc.markNodeError(retryNodeName, err), err
 		}
+		if lastChildNode != nil && lastChildNode.Pending() && processedTmpl.GetType() == wfv1.TemplateTypeContainer {
+			_, err := woc.createWorkflowPod(lastChildNode.Name, *processedTmpl.Container, processedTmpl, false)
+			if apierr.IsForbidden(err) {
+				return woc.markNodePending(lastChildNode.Name, err), nil
```

Comment:
This return statement is incorrect. We should only return the `NodeStatus` of the parent node here (`retryParentNode`), not of the child node. The code as is would be a large bug.